client/lib/i18n-utils/utils.js

```diff
@@ -54,6 +54,30 @@ const i18nUtils = {
 	},
 
 	/**
+	 * Given the content of an 'AcceptedLanguages' request header, returns an array of the languages.
+	 *
+	 * This differs slightly from other language functions, as it doesn't try to validate the language codes,
+	 * or merge similar language codes.
+	 *
+	 * @param  {string} header - The content of the AcceptedLanguages header.
+	 * @return {Array} An array of language codes in the header, all in lowercase.
+	 */
+	getAcceptedLanguagesFromHeader: function( header ) {
+		if ( ! header ) {
+			return [];
+		}
+
+		return filter( map( header.split( ',' ), lang => {
+			const match = lang.match( /^[A-Z]{2,3}(-[A-Z]{2,3})?/i );
+			if ( ! match ) {
+				return false;
```

Comment:
Shouldn't we return an empty array instead here?

Reply:
Nope, that's returning from the `map()` filter, so `false` will remove the item from the array.